frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/EmbedModalContent.tsx

```diff
@@ -0,0 +1,234 @@
+        parameter.slug,
+        parameterValues[parameter.id] ?? null,
+      ]),
+    );
+  };
+
+  const previewParametersBySlug = getPreviewParamsBySlug();
+  const previewParameters = getPreviewParameters(
+    resourceParameters,
+    embeddingParams,
+  );
+
+  if (!embedType) {
+    return (
+      <SelectEmbedTypePane
+        resource={resource}
+        resourceType={resourceType}
+        onCreatePublicLink={onCreatePublicLink}
+        onDeletePublicLink={onDeletePublicLink}
+        getPublicUrl={getPublicUrl}
+        onChangeEmbedType={setEmbedType}
+      />
+    );
+  }
+
+  const hasSettingsChanges = !_.isEqual(
+    resource.embedding_params,
+    embeddingParams,
+  );
+
+  return (
+    <div className="flex flex-column full-height">
+      <EmbedModalContentStatusBar
+        resourceType={resourceType}
+        isEmbeddingEnabled={resource.enable_embedding}
+        hasSettingsChanges={hasSettingsChanges}
+        onSave={handleSave}
+        onUnpublish={handleUnpublish}
+        onDiscard={handleDiscard}
+      />
+
+      <div className="flex flex-full">
+        <StaticEmbedSetupPane
+          activePane={pane}
+          resource={resource}
+          resourceType={resourceType}
+          embedType={embedType}
+          token={getSignedToken(
+            resourceType,
+            resource.id,
+            previewParametersBySlug,
+            secretKey,
+            embeddingParams,
+          )}
+          iframeUrl={getSignedPreviewUrl(
+            siteUrl,
+            resourceType,
+            resource.id,
+            previewParametersBySlug,
+            displayOptions,
+            secretKey,
+            embeddingParams,
+          )}
+          siteUrl={siteUrl}
+          secretKey={secretKey}
+          params={previewParametersBySlug}
+          displayOptions={displayOptions}
+          previewParameters={previewParameters}
+          parameterValues={parameterValues}
+          resourceParameters={resourceParameters}
+          embeddingParams={embeddingParams}
+          onChangeDisplayOptions={setDisplayOptions}
+          onChangeEmbeddingParameters={setEmbeddingParams}
+          onChangeParameterValue={(id: string, value: string) =>
+            setParameterValues(prevState => ({
+              ...prevState,
+              [id]: value,
+            }))
+          }
+          onChangePane={setPane}
+        />
+      </div>
+    </div>
+  );
+};
+
+function getDefaultEmbeddingParams(
+  resource: EmbedResource,
+  resourceParameters: EmbedResourceParameter[],
+) {
+  return filterValidResourceParameters(
+    resourceParameters,
+    resource.embedding_params || {},
+  );
+}
+
+function filterValidResourceParameters(
+  resourceParameters: EmbedResourceParameter[],
+  embeddingParams: EmbeddingParameters,
+) {
+  const validParameters = resourceParameters.map(parameter => parameter.slug);
+
+  return _.pick(embeddingParams, validParameters);
+}
+
+function getPreviewParameters(
```

Comment:
Since the UI says _Preview locked parameters_
![image](https://github.com/metabase/metabase/assets/1937582/3715f6ef-a1c4-4f2d-9289-d03cbee404e8)

I think we should name this `getPreviewLockedParameters` (the same as the copy) or just `getLockedParameters` (the same as the implementation).

Naming this just `getPreviewParameters` kind of confused me at first because I don't remember that the only previewable parameter is the locked one.

This comment applies to other variables that are named `*previewParameters*`